assets/js/modules/adsense/components/setup/v2/SetupMain.js

```diff
@@ -0,0 +1,319 @@
+	determineAccountStatus,
+	determineClientID,
+	determineSiteStatus,
+} from '../../../util/status';
+const { useSelect, useDispatch } = Data;
+
+export default function SetupMain() {
+	const viewContext = useContext( ViewContextContext );
+	const eventCategory = `${ viewContext }_adsense`;
+
+	const {
+		clearError,
+		resetAccounts,
+		resetAlerts,
+		resetClients,
+		resetSites,
+		resetURLChannels,
+		setAccountID,
+		setAccountStatus,
+		submitChanges,
+	} = useDispatch( MODULES_ADSENSE );
+
+	const [
+		isAwaitingBackgroundSubmit,
+		setIsAwaitingBackgroundSubmit,
+	] = useState( false );
+	// Submit changes for determined parameters in the background when they are valid.
+	const [ isSubmittingInBackground, setIsSubmittingInBackground ] = useState(
+		false
+	);
+
+	const siteURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getReferenceSiteURL()
+	);
+	const isAdBlockerActive = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).isAdBlockerActive()
+	);
+	const accounts = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getAccounts()
+	);
+	const previousAccountID = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getAccountID()
+	);
+	const previousClientID = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getClientID()
+	);
+	const accountsError = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getError( 'getAccounts', [] )
+	);
+	// Check whether settings differ from server and are valid.
+	const canSubmitChanges = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).canSubmitChanges()
+	);
+
+	const accountID = determineAccountID( {
+		accounts,
+		previousAccountID,
+	} );
+
+	const clients = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getClients( accountID )
+	);
+	const alerts = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getAlerts( accountID )
+	);
+	const alertsError = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getError( 'getAlerts', [ accountID ] )
+	);
+
+	const clientID = determineClientID( {
+		clients,
+		previousClientID,
+	} );
+	const urlChannels = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getURLChannels( accountID, clientID )
+	);
+	const urlChannelsError = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getErrorForSelector( 'getURLChannels', [
+			accountID,
+			clientID,
+		] )
+	);
+
+	const accountStatus = determineAccountStatus( {
+		accounts,
+		clients,
+		alerts,
+		urlChannels,
+		accountsError,
+		alertsError,
+		urlChannelsError,
+		previousAccountID,
+		previousClientID,
+	} );
+	const siteStatus = determineSiteStatus( {
+		urlChannels,
+		siteURL,
+	} );
+
+	useUpdateEffect( () => {
+		if (
+			accountID !== undefined ||
+			clientID !== undefined ||
+			accountStatus !== undefined ||
+			siteStatus !== undefined
+		) {
+			setIsAwaitingBackgroundSubmit( true );
+		}
+	}, [ accountID, clientID, accountStatus, siteStatus ] );
+
+	// Update current account ID setting on-the-fly.
+	useEffect( () => {
+		if (
+			! accountID ||
```

Comment:
🤔 @eugene-manuilov actually there's an early return for that condition, basically we are saying do not to anything if accountID is empty/falsy.

Reply:
@asvinb AC says "_If there is only one account in the returned list from the API and **no accountID is set yet**._" which means that we should return early if accountID **is set**. That's why i think it should be `!! accountID` instead of `! accountID`. Does it make sense to you?

Reply:
Ah... I think you are right... Sorry for confusion.